ext/spl/spl_array.c

```diff
@@ -783,11 +780,18 @@ static inline HashTable* spl_array_get_debug_info(zend_object *obj) /* {{{ */
 		storage = &intern->array;
 		Z_TRY_ADDREF_P(storage);
 
-		base = obj->handlers == &spl_handler_ArrayIterator
+		const zend_class_entry *base_class_ce = instanceof_function(obj->ce, spl_ce_ArrayIterator)
 			? spl_ce_ArrayIterator : spl_ce_ArrayObject;
-		zname = spl_gen_private_prop_name(base, "storage", sizeof("storage")-1);
-		zend_symtable_update(debug_info, zname, storage);
-		zend_string_release_ex(zname, 0);
+
+		zend_string *mangled_named = zend_mangle_property_name(
```

Comment:
IMO there was nothing wrong with the old code having a specialised helper, now there's more repetition in the code. I would prefer this commit to be dropped.

Reply:
In that case I would prefer an API that actually sets the hashtable value at the same time, rather than just get the mangled name.

I'll see what I can cook up.